doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -145,34 +138,60 @@ public void updateDoiRequest(UUID publicationID, DoiRequestStatus requestedStatu
         putItem(publication);
     }
 
+    private boolean belongsToUser(String user, String publicationOwner) {
+        return nonNull(publicationOwner) && publicationOwner.equals(user);
+    }
+
+    private List<Publication> fetchPublicationsByPublisherAndStatus(URI publisher, DoiRequestStatus status)
```

Comment:
All this code has the weirdness of not simply telling the object what to do, you extract information and you have to combine and tweak around with data. 

Maybe doesnt belong in `DynamoDBDoiRequestService` which seems to be a storage service interface for accessing doi status stored in dynamodb.  

`Publication` has 2 mixed roles with this, one for being business object and one for being a DTO for the Dynamo backend? 

Responsibility concern comes  into mind.

Reply:
One of the things that will happen in the next PR is that we will remove some classes and we will use Publication for DTO for now (in the scope of the bugfix). We do need to define the way to construct DTOs DAOs and BOs but this is outside the scope of this bugifx.

Reply:
Created a bug for class responsibility problem: https://unit.atlassian.net/browse/NP-1573